lib/sqlalchemy/orm/mapped_collection.py

```diff
@@ -381,12 +436,14 @@ def remove(self, value, _sa_initiator=None):
                 "based on mutable properties or properties that only obtain "
                 "values after flush?" % (value, self[key], key)
             )
-        self.__delitem__(key, _sa_initiator)
+        self.__delitem__(key, _sa_initiator)  # type: ignore[call-arg]
 
 
-def _mapped_collection_cls(keyfunc, ignore_unpopulated_attribute):
-    class _MKeyfuncMapped(KeyFuncDict):
-        def __init__(self):
+def _mapped_collection_cls(
+    keyfunc: Callable[[_KT], _KT], ignore_unpopulated_attribute: bool
+) -> type[KeyFuncDict[_KT, _KT]]:
+    class _MKeyfuncMapped(KeyFuncDict):  # type: ignore
```

Comment:
I assume that return type was caught in runtime with PyAnnotate, and is correct. If so, couldn't we just put the same type arg signature `[_KT, _KT]` on the class definiton? Mypy complains that it has no type args, however both `_KT` and `_VT` are bound to `Any`, so essentially any of them will remove the error. I am not sure about the correct combination, though, - we could trust mypy for now, and then just leave a comment for Mike.
```suggestion
def _mapped_collection_cls(
    keyfunc: Callable[[_KT], _KT], ignore_unpopulated_attribute: bool
) -> type[KeyFuncDict[_KT, _KT]]:
    class _MKeyfuncMapped(KeyFuncDict[_KT, _KT]):
```